curp/src/test_utils/execute_error.rs

```diff
@@ -0,0 +1,23 @@
+use std::io;
+
+use thiserror::Error;
+
+/// Error met when executing commands
+#[allow(clippy::module_name_repetitions)] // this-error generate code false-positive
+#[non_exhaustive]
+#[derive(Error, Debug, Clone)]
+pub(crate) enum ExecuteError {
+    /// Command is invalid
+    #[error("invalid command {0} ")]
+    InvalidCommand(String),
+    /// Met I/O error while executing
+    #[error("meet io related error")]
+    IoError(String),
+}
+
+impl From<io::Error> for ExecuteError {
+    #[inline]
+    fn from(err: io::Error) -> Self {
+        Self::IoError(err.to_string())
+    }
+}
```

Comment:
I guess there is no need to have such a complex execute error for test command. String would be fine.